### PR TITLE
Extends SvgTheme to have locale and decideFontFamily.

### PR DESCRIPTION
**Pull request by espresso3389**
For #688, I extended `SvgTheme` to have `locale` and also `decideFontFamily` callback that decide actual font from list of font families listed on `text` element.

This is just a draft PR but I want to discuss how to deal with such cases.

### CODE REVIEW

File: lib/src/svg/parser_state.dart [hunk 1]
```diff
@@ -1783,6 +1785,57 @@ class SvgParserState {
 
     throw StateError('Could not parse "$colorString" as a color.');
   }
+
+  /// Parse CSS style font family list to decide a font family.
+  String? parseFontFamily(String? fontFamily) {
+    final fontFamilies = _parseStringList(fontFamily);
+    if (theme.resolveFontFamily != null) {
+      return theme.resolveFontFamily!(fontFamilies);
```
Comment (dnfield) [inline lib/src/svg/parser_state.dart:1793]: What's the use case here?

Why would we avoid using the `fontFamilyFallack` on `TextStyle` to append any additional fonts?

File: lib/src/svg/parsers.dart [hunk 2]
```diff
@@ -191,8 +191,9 @@ Paragraph createParagraph(
   String text,
   DrawableStyle style,
   DrawablePaint? foregroundOverride,
+  Locale? locale,
 ) {
-  final ParagraphBuilder builder = ParagraphBuilder(ParagraphStyle())
+  final ParagraphBuilder builder = ParagraphBuilder(ParagraphStyle(locale: locale))
```
Comment (dnfield) [inline lib/src/svg/parsers.dart:196]: Perhaps
```suggestion
  final ParagraphBuilder builder = ParagraphBuilder(ParagraphStyle(locale: locale ?? PlatformDispatcher.instance.locale))
```

File: lib/src/svg/parser_state.dart (hunk 1 above)
Comment (espresso3389) [reply lib/src/svg/parser_state.dart:1793]: Yes, appending them to `fontFamilyFallack` may help but `font-family` on `text` may have many variations and they cannot be recognized by the system... How should I...

For my own use case, a SVG file that has

```html
<text fill="#fff" font-family="YuGo-Medium-83pv-RKSJ-H, YuGothic" font-size="4px" transform="translate(6.63 22.25)">
```

There are two font names here; `YuGo-Medium-83pv-RKSJ-H` and `YuGothic`.

And, it seems that iOS/macOS can resolve the font name `YuGothic` but Windows can't. There should be `Yu Gothic`.

Therefore, in my app, I just made a hand-crafted mapping table and translate such variations.

Comment (dnfield) [reply lib/src/svg/parser_state.dart:1793]: Why not update your SVG to include `Yu Gothic` in the `font-family`? :)

Comment (espresso3389) [reply lib/src/svg/parser_state.dart:1794]: Yes, it's sometimes the correct and best way to update SVG to include both "YuGothic", "Yu Gothic", and also "游ゴシック". But it's not acceptable if we have many files...

Comment (espresso3389) [reply lib/src/svg/parser_state.dart:1794]: Anyway, I prefer some dynamic way to resolve font names based on installed fonts or platform name, or... So it's better to equip a away to resolve such kind of mechanism.

**Comment by dnfield** [comment]
This PR is out of date with the main branch, closing for now.

**Pull request closed without merging**